src/main/java/zd/zero/waifu/motivator/plugin/assets/TextAssetService.kt

```diff
@@ -2,34 +2,66 @@ package zd.zero.waifu.motivator.plugin.assets
 
 import com.google.gson.Gson
 import com.google.gson.reflect.TypeToken
-import java.net.URI
+import com.intellij.openapi.diagnostic.Logger
+import zd.zero.waifu.motivator.plugin.tools.ExceptionTools.runSafely
+import zd.zero.waifu.motivator.plugin.tools.toOptional
+import java.io.BufferedReader
 import java.nio.file.Files
-import java.nio.file.Paths
+import java.util.*
 import java.util.concurrent.ConcurrentHashMap
 import kotlin.random.Random
 
 object TextAssetService {
     private val gson = Gson()
+    private val log: Logger? = Logger.getInstance(this::class.java)
     private val categoryToTextDefinitions: MutableMap<WaifuAssetCategory, List<TextualMotivationAsset>> =
-            ConcurrentHashMap()
+        ConcurrentHashMap()
     private val random = Random(System.currentTimeMillis())
 
-    fun pickRandomAssetByCategory(waifuAssetCategory: WaifuAssetCategory): TextualMotivationAsset {
-        val listOfAssets = getListOfAssets(waifuAssetCategory)
-        return listOfAssets.random(random)
-    }
+    fun pickRandomAssetByCategory(waifuAssetCategory: WaifuAssetCategory): Optional<TextualMotivationAsset> =
+        getListOfAssets(waifuAssetCategory)
+            .filter { it.isNotEmpty() }
+            .map {
+                it.random(random)
+            }
 
-    private fun getListOfAssets(waifuAssetCategory: WaifuAssetCategory): List<TextualMotivationAsset> =
+    private fun getListOfAssets(waifuAssetCategory: WaifuAssetCategory): Optional<List<TextualMotivationAsset>> =
         if (categoryToTextDefinitions.containsKey(waifuAssetCategory)) {
-            categoryToTextDefinitions[waifuAssetCategory]!!
+            categoryToTextDefinitions[waifuAssetCategory]!!.toOptional()
         } else {
-            Files.newBufferedReader(
-                    Paths.get(URI(TextAssetDefinitionService.getRandomAssetByCategory(waifuAssetCategory).path))
+            TextAssetDefinitionService.getRandomAssetByCategory(waifuAssetCategory)
+                .flatMap { assetPackage ->
+                    attemptToReadAsset(assetPackage)
+                }
+                .map {
+                    categoryToTextDefinitions[waifuAssetCategory] = it
```

Comment:
Lol, I forgot to save the read file contents to cache, so it always read from the file and returned that.

I wrote some tests to make sure that it only reads once.